DiagnosticsExtension/Models/ConnectionStringValidator/BlobStorageValidator.cs

```diff
@@ -0,0 +1,196 @@
+            }
+
+            return response;
+        }
+
+        public async Task<TestConnectionData> TestConnectionStringViaAppSetting(string appSettingName, string entityName)
+        {
+            var envDict = Environment.GetEnvironmentVariables();
+            string appSettingClientIdValue, appSettingClientCredValue = null;
+            string value = null;
+            BlobServiceClient client = null;
+
+            if (envDict.Contains(appSettingName))
+            {
+                try
+                {
+                    value = Environment.GetEnvironmentVariable(appSettingName);
+                    client = new BlobServiceClient(value);
+                }
+                catch (ArgumentNullException e)
+                {
+                    throw new EmptyConnectionStringException(e.Message, e);
+                }
+                catch (Exception e)
+                {
+                    throw new MalformedConnectionStringException(e.Message, e);
+                }
+            }
+            else
+            {
+                try
+                {
+
+
+                    value = Environment.GetEnvironmentVariable(appSettingName + "__blobServiceUri");
+                    if (string.IsNullOrEmpty(value))
+                    {
+                        value = Environment.GetEnvironmentVariable(appSettingName + "__serviceUri");
+                    }
+                    if (!string.IsNullOrEmpty(value))
+                    {
+                        appSettingClientIdValue = Environment.GetEnvironmentVariable(appSettingName + "__clientId");
+                        appSettingClientCredValue = Environment.GetEnvironmentVariable(appSettingName + "__credential");
+                        Uri objuri = new Uri(value);
+                        if (!string.IsNullOrEmpty(appSettingClientIdValue))
+                        {
+                            if (appSettingClientCredValue != "managedidentity")
+                            {
+                                throw new ManagedIdentityException("managedidentitymissed");
+                            }
+                            else
+                            {
+                                identityType = ConnectionStringValidationResult.ManagedIdentityType.User;
+                                client = new BlobServiceClient(objuri, new Azure.Identity.ManagedIdentityCredential(appSettingClientIdValue));
+                            }
+                        }
+                        else
+                        {
+                            identityType = ConnectionStringValidationResult.ManagedIdentityType.System;
+                            client = new BlobServiceClient(objuri, new Azure.Identity.ManagedIdentityCredential());
+                        }
+                    }
+                    else
+                    {
+                        throw new ManagedIdentityException("fullyQualifiedNamespacemissed");
```

Comment:
see comment about defining a const string.  Alternately you could simply use the string value of the enum ResultStatus.FullyQualifiedNamespacemissed to set the exception text

Reply:
fixed